pixeltable/utils/pydantic.py

```diff
@@ -0,0 +1,62 @@
+import datetime
+import typing
+from enum import Enum
+from typing import Any, Union
+
+import pydantic
+
+
+def is_json_convertible(model: type[pydantic.BaseModel]) -> bool:
+    """
+    Determine if instances of a Pydantic model can be converted to valid JSON
+    based on the type hints of its fields.
+    """
+    type_hints = typing.get_type_hints(model)
+    return all(_type_is_json_convertible(field_type) for field_type in type_hints.values())
+
+
+def _type_is_json_convertible(type_hint: Any) -> bool:
+    """Recursively check if a type hint represents a JSON-compatible type."""
+    if type_hint is type(None):
+        return True
+    if type_hint is Any:
+        return False
+
+    if type_hint in (str, int, float, bool):
+        return True
+
+    # datetime types serialize to ISO strings
+    if type_hint is datetime.datetime or type_hint is datetime.date:
```

Comment:
can't this be merged with the previous line?

`if type_hint in (str, int, float, bool, datetime.datetime, datetime.date)`